### PR TITLE
Deprecate 'slient' param in get_blocks_()

### PR DESCRIPTION
**Pull request by IvanaGyro**
"slient" argument was corrected to "silent" in #666. However, that change broke the current Python API. This PR adds "slient" back and marks it deprecated until v2.0.0.

- Restore 'slient' parameter support to fix API breaking change from previous commit
- Add FutureWarning for deprecated 'slient' parameter using py::warnings
- Implement robust argument validation with detailed error messages
- Bump pybind11 requirement to >=3.0 (py::warnings available since pybind11 3.0)
- Add comprehensive test suite covering deprecation warnings, argument validation, and edge cases

### CODE REVIEW

File: pybind/unitensor_py.cpp [hunk 1]
```diff
@@ -700,11 +701,54 @@ void unitensor_binding(py::module &m) {
...
+          silent = py::cast<bool>(args[0]);
+        } else if (kwargs.contains("slient")) {
+            py::warnings::warn(
+                "Keyword 'slient' is deprecated and will be removed in v2.0.0; use 'silent' instead.",
+                PyExc_FutureWarning, 2);
+            silent = kwargs["slient"].cast<bool>();
+        } else if (kwargs.contains("silent")) {
+            silent = kwargs["silent"].cast<bool>();
+        } else if (kwargs.size() == 1) {
+            // The case that kwargs.size() > 1 has been catched above.
```
Comment (jysh1214) [inline pybind/unitensor_py.cpp:744]: typo: catched -> caught

**Review by jysh1214 — APPROVED** [review]
LGTM.
